File: chain-sdk/clojure/src/on_chain_api/specs/token_v3_dto.clj
(ns on-chain-api.specs.token-v3-dto
  (:require [clojure.spec.alpha :as s]
            [spec-tools.data-spec :as ds]
            [on-chain-api.specs.big-integer :refer :all]
            [on-chain-api.specs.big-integer :refer :all]
            [on-chain-api.specs.big-integer :refer :all]
            )
  (:import (java.io File)))


(def token-v3-dto-data
  {
   (ds/opt :entry_time) inst?
   (ds/opt :recv_time) inst?
   (ds/opt :block_number) int?
   (ds/opt :vid) int?
   (ds/opt :id) string?
   (ds/opt :symbol) string?
   (ds/opt :name) string?
   (ds/opt :decimals) int?
   (ds/opt :total_supply) big-integer-spec
   (ds/opt :volume) string?
   (ds/opt :volume_usd) string?
   (ds/opt :untracked_volume_usd) string?
   (ds/opt :fees_usd) string?
   (ds/opt :tx_count) big-integer-spec
   (ds/opt :pool_count) big-integer-spec
   (ds/opt :total_value_locked) string?
   (ds/opt :total_value_locked_usd) string?
   (ds/opt :total_value_locked_usd_untracked) string?
   (ds/opt :derived_eth) string?
   (ds/opt :whitelist_pools) (s/coll-of string?)
   (ds/opt :token_symbol) string?
   })

(def token-v3-dto-spec
  (ds/spec
    {:name ::token-v3-dto
     :spec token-v3-dto-data}))

File: chain-sdk/javascript-closure-angular/API/Client/TokenV3DTO.js
goog.provide('API.Client.TokenV3DTO');

/**
 * Stores aggregated information for a specific token across all pairs that token is included in.
 * @record
 */
API.Client.TokenV3DTO = function() {}

/**
 * @type {!Date}
 * @export
 */
API.Client.TokenV3DTO.prototype.entryTime;

/**
 * @type {!Date}
 * @export
 */
API.Client.TokenV3DTO.prototype.recvTime;

/**
 * 
 * @type {!number}
 * @export
 */
API.Client.TokenV3DTO.prototype.blockNumber;

/**
 * 
 * @type {!number}
 * @export
 */
API.Client.TokenV3DTO.prototype.vid;

/**
 * token address
 * @type {!string}
 * @export
 */
API.Client.TokenV3DTO.prototype.id;

/**
 * token symbol
 * @type {!string}
 * @export
 */
API.Client.TokenV3DTO.prototype.symbol;

/**
 * token name
 * @type {!string}
 * @export
 */
API.Client.TokenV3DTO.prototype.name;

/**
 * token decimals
 * @type {!number}
 * @export
 */
API.Client.TokenV3DTO.prototype.decimals;

/**
 * @type {!API.Client.BigInteger}
 * @export
 */
API.Client.TokenV3DTO.prototype.totalSupply;

/**
 * volume in token units
 * @type {!string}
 * @export
 */
API.Client.TokenV3DTO.prototype.volume;

/**
 * volume in derived USD
 * @type {!string}
 * @export
 */
API.Client.TokenV3DTO.prototype.volumeUsd;

/**
 * volume in USD even on pools with less reliable USD values
 * @type {!string}
 * @export
 */
API.Client.TokenV3DTO.prototype.untrackedVolumeUsd;

/**
 * fees in USD
 * @type {!string}
 * @export
 */
API.Client.TokenV3DTO.prototype.feesUsd;

/**
 * @type {!API.Client.BigInteger}
 * @export
 */
API.Client.TokenV3DTO.prototype.txCount;

/**
 * @type {!API.Client.BigInteger}
 * @export
 */
API.Client.TokenV3DTO.prototype.poolCount;

/**
 * liquidity across all pools in token units
 * @type {!string}
 * @export
 */
API.Client.TokenV3DTO.prototype.totalValueLocked;

/**
 * liquidity across all pools in derived USD
 * @type {!string}
 * @export
 */
API.Client.TokenV3DTO.prototype.totalValueLockedUsd;

/**
 * TVL derived in USD untracked
 * @type {!string}
 * @export
 */
API.Client.TokenV3DTO.prototype.totalValueLockedUsdUntracked;

/**
 * derived price in ETH
 * @type {!string}
 * @export
 */
API.Client.TokenV3DTO.prototype.derivedEth;

/**
 * pools token is in that are white listed for USD pricing
 * @type {!Array<!string>}
 * @export
 */
API.Client.TokenV3DTO.prototype.whitelistPools;

/**
 * @type {!string}
 * @export
 */
API.Client.TokenV3DTO.prototype.tokenSymbol;

